apps/full-site-editing/full-site-editing-plugin/full-site-editing/blocks/post-content/edit.js

```diff
@@ -11,94 +13,137 @@ import { get } from 'lodash';
 import { IconButton, Placeholder, Toolbar } from '@wordpress/components';
 import { compose, withState } from '@wordpress/compose';
 import { withSelect } from '@wordpress/data';
-import { BlockControls } from '@wordpress/editor';
-import { Fragment, RawHTML } from '@wordpress/element';
+import { BlockControls, InnerBlocks, PostTitle } from '@wordpress/editor';
+import { Component, Fragment, RawHTML } from '@wordpress/element';
 import { __, sprintf } from '@wordpress/i18n';
 
 /**
  * Internal dependencies
  */
 import PostAutocomplete from '../../components/post-autocomplete';
 
-const PostContentEdit = compose(
-	withState( {
-		isEditing: false,
-		selectedPostId: undefined,
-		selectedPostType: undefined,
-	} ),
-	withSelect( ( select, { selectedPostId, selectedPostType } ) => {
-		const { getEntityRecord } = select( 'core' );
-		return {
-			selectedPost: getEntityRecord( 'postType', selectedPostType, selectedPostId ),
-		};
-	} )
-)( ( { attributes, isEditing, selectedPost, setState } ) => {
-	const { align } = attributes;
+class PostContentEdit extends Component {
+	componentDidMount() {
+		this.toggleEditorPostTitleVisibility();
+	}
 
-	const toggleEditing = () => setState( { isEditing: ! isEditing } );
+	componentDidUpdate( prevProps ) {
+		if ( this.props.isFullSitePage !== prevProps.isFullSitePage ) {
+			this.toggleEditorPostTitleVisibility();
+		}
+	}
 
-	const onSelectPost = ( { id, type } ) => {
-		setState( {
+	componentWillUnmount() {
+		this.toggleEditorPostTitleVisibility( true );
+	}
+
+	toggleEditing() {
+		const { isEditing, setState } = this.props;
+		setState( { isEditing: ! isEditing } );
+	}
+
+	onSelectPost( { id, type } ) {
+		this.props.setState( {
 			isEditing: false,
 			selectedPostId: id,
 			selectedPostType: type,
 		} );
-	};
+	}
+
+	/**
+	 * Hides the default post title of the editor when editing a full site page and shows new post title rendered by the
+	 * post content block in order to have it just before the content of the post.
+	 *
+	 * @param {boolean} forceDefaultPostTitle Whether the default post title should be always displayed.
+	 */
+	toggleEditorPostTitleVisibility( forceDefaultPostTitle = false ) {
+		const showPostTitleBeforeContent = this.props.isFullSitePage && ! forceDefaultPostTitle;
+		document
+			.querySelector( '#editor' )
+			.classList.toggle( 'show-post-title-before-content', showPostTitleBeforeContent );
```

Comment:
Let's guard if the querySelector returns `null` for no results. In that case we may throw `Cannot read property 'classList' of null`

Reply:
🤔 on second thought is it possible to conditionally add a class or enqueue CSS when we match this case in a filter? That might be a bit more robust and would avoid any potential rendering flashes.

cc @kwight

Reply:
Good suggestion! That will also remove the need of creating a new selector, simplifying the approach here.

But we should investigate in a follow up if it's possible to make a block unremovable. Otherwise, the user can get an editor without any post title if they remove the post content block since the class added by the filter would keep the normal post title hidden.